modules/caddyhttp/caddyauth/command.go

```diff
@@ -67,17 +68,28 @@ func cmdHashPassword(fs caddycmd.Flags) (int, error) {
 	salt := []byte(fs.String("salt"))
 
 	if len(plaintext) == 0 {
-		if terminal.IsTerminal(int(os.Stdin.Fd())) {
-			fmt.Print("Enter password: ")
-			plaintext, err = terminal.ReadPassword(int(os.Stdin.Fd()))
-			fmt.Println()
+		fd := int(os.Stdin.Fd())
+		if terminal.IsTerminal(fd) {
+			// ensure the terminal state is restored on SIGINT
+			state, _ := terminal.GetState(fd)
+			c := make(chan os.Signal)
+			signal.Notify(c, os.Interrupt)
+			go func() {
+				<-c
+				_ = terminal.Restore(fd, state)
+				os.Exit(caddy.ExitCodeFailedStartup)
+			}()
```

Comment:
I think this goroutine leaks if no signal is received.

Reply:
:thinking: I don't know what that means frankly. Still a relative noob at Go :smile:

Reply:
@mholt what should I do here? :thinking:

Reply:
I guess that's true, it's just not as clean as it could be, probably.

What I mean is that `<-c` never unblocks if the program never receives an interrupt. So if the user goes through the process normally, this goroutine will never exit. Of course, as you note, this is a short-lived command, I'm just noting it's not the cleanest Go code since it does technically leak a goroutine.

If we're going to leave it unfixed, we should at least comment to explain that the goroutine never unblocks (feel free to verify by adding `log.Println()` on both sides of the `<-c` and see what happens) but that we don't care because the program terminates quickly anyway.

A proper fix probably involves closing the channel (`close(c)`) when we're no longer accepting signals... oh -- actually maybe https://golang.org/pkg/os/signal/#Stop needs to be used (instead?).

Reply:
I added `defer signal.Stop(c)`, does what we need I think.